### PR TITLE
345 align find one by id in repositories

### LINKED ISSUE #345: Align find_one_by_id in repositories
We've added find_one_by_id_optional to repositories that already have find_one_by_id that returned repository error (to reduce size of PR that use new 'optional' return). We need to consolidate this into find_one_by_id that returns optional for all repositories. I've tried to do this in this pr: https://github.com/openmsupply/open-msupply/pull/245#discussion_r922940280, but ended up breaking quite a lot of tests. I think the best way to do it, is to remove find_one_by_id for repositories that return record not found as error, then fix compilation errors by changing to _optional (make sure tests pass), and then change form _optional to just find_one_by_id.

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #345 

# 👩🏻‍💻 What does this PR do?
Refactor all find_one_by_id to return an Optional row.

## 💌 Any notes for the reviewer?

# 🧪 Testing

- [ ] Make sure tests are still running

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [x] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: server/repository/src/db_diesel/invoice_line_row.rs [hunk 1]
```diff
@@ -48,21 +48,16 @@ joinable!(invoice_line -> return_reason (return_reason_id));
 allow_tables_to_appear_in_same_query!(invoice_line, item_link);
 allow_tables_to_appear_in_same_query!(invoice_line, name_link);
 
-#[derive(DbEnum, Debug, Clone, PartialEq, Eq)]
+#[derive(DbEnum, Debug, Clone, PartialEq, Eq, Default)]
 #[DbValueStyle = "SCREAMING_SNAKE_CASE"]
 pub enum InvoiceLineType {
+    #[default]
```
Comment (roxy-dao) [inline server/repository/src/db_diesel/invoice_line_row.rs:54]: Side quest~

File: server/repository/src/repository_error.rs [hunk 2]
```diff
@@ -11,7 +11,7 @@ pub enum RepositoryError {
     /// Foreign key constraint is violated
     #[error("foreign key constraint is violated {0}")]
     ForeignKeyViolation(String),
-    /// Actix thred pool canceled
+    /// Actix thread pool cancelled
```
Comment (roxy-dao) [inline server/repository/src/repository_error.rs:14]: Side quest~

File: server/service/src/invoice/inbound_return/insert/validate.rs [hunk 3]
```diff
@@ -14,10 +11,9 @@ pub fn validate(
     input: &InsertInboundReturn,
 ) -> Result<Name, InsertInboundReturnError> {
     use InsertInboundReturnError::*;
-    check_invoice_does_not_exists(&input.id, connection).map_err(|e| match e {
-        InvoiceAlreadyExistsError::InvoiceAlreadyExists => InvoiceAlreadyExists,
-        InvoiceAlreadyExistsError::RepositoryError(err) => DatabaseError(err),
-    })?;
+    if let Some(_) = check_invoice_exists(&input.id, connection)? {
```
Comment (roxy-dao) [inline server/service/src/invoice/inbound_return/insert/validate.rs:14]: Got rid of check_invoice_does_not_exist method and just reused this one to return error if it is Some

**Review by jmbrunskill — APPROVED** [review]
Looks good to me, haven't tested anything in the frontend. Can't see it causing problems.

File: server/service/src/invoice/inbound_return/insert/validate.rs (hunk 3 above)
Comment (jmbrunskill) [reply server/service/src/invoice/inbound_return/insert/validate.rs:14]: Nice!

**Pull request merged**